src/file_dialog.rs

```diff
@@ -2209,70 +2270,92 @@ impl FileDialog {
         let mut should_return = false;
 
         ui.with_layout(egui::Layout::top_down_justified(egui::Align::LEFT), |ui| {
-            let scroll_area = egui::containers::ScrollArea::vertical().auto_shrink([false, false]);
+            let shift_modifier = ui.input(|i| i.modifiers.shift_only());
+            let command_modifier = ui.input(|i| i.modifiers.command);
+            let shift_only_modifier = ui.input(|i| i.modifiers.shift_only());
+
+            let row_height = Self::get_row_height(ui);
+            let table = TableBuilder::new(ui)
+                .sense(egui::Sense::click())
+                .striped(true)
+                .resizable(true)
+                .cell_layout(egui::Layout::left_to_right(egui::Align::Center))
+                .column(Column::auto().at_least(120.0)) // "Name"
+                .column(Column::auto().at_least(70.0)) // "File Size"
+                .column(Column::auto().at_least(60.0)) // "Date Created"
+                .column(Column::remainder().at_least(60.0)) // "Date Modified"
+                .header(row_height, |mut header| {
+                    self.add_sortable_column(&mut header, "Name", SortBy::Filename, &mut data);
+                    self.add_sortable_column(&mut header, "File Size", SortBy::Size, &mut data);
+                    self.add_sortable_column(
+                        &mut header,
+                        "Created",
+                        SortBy::DateCreated,
+                        &mut data,
+                    );
+                    self.add_sortable_column(
+                        &mut header,
+                        "Modified",
+                        SortBy::DateLastModified,
+                        &mut data,
+                    );
```

Comment:
All these labels ("Name", "File Size", "Created", "Modified") should be configurable via `FileDialogLabels`

Reply:
Done, but now the function has too many lines:

```rust
 #[allow(clippy::too_many_lines)]
fn ui_update_central_panel_content(&mut self, ui: &mut egui::Ui) {
    // ...
}
```

Reply:
not really, it's already divided. but the functions take a lot of arguments, so it is split on many lines...